### DISCUSSION TITLE
[SUGGESTION] Move zha-toolkit repository to zigpy organization?

### DISCUSSION THREAD

**Discussion by mdeweerd** [zigpy/zigpy#901] [ZHA development]
Following the message from @Hedda : https://github.com/mdeweerd/zha-toolkit/issues/3.  It's well said, so I am not rephrasing him ;-) .

My response: I am fine with "moving" zha-toolkit.  If this entails any restrictions/rules - let me know up front.

**Comment by Hedda** [comment]
Like to bump this suggestion/question. zha-toolkit provides great values via low-level and high-level Zigbee commands through ZHA/zigpy:

https://community.home-assistant.io/t/zha-toolkit-a-big-set-of-zigbee-commands-on-top-of-zha-zigpy/373346

**[@mdeweerd](https://github.com/mdeweerd)** can I suggest that you consider asking/discussing the possibility of having your zha-toolkit repository moved from https://github.com/mdeweerd/zha-toolkit moved to the zigpy organization @ https://github.com/zigpy/ on GitHub so that the mainstream copy of this repo will stay there become upstream for shared access together by the collective of developers that make up the zigpy organization?
 
Note that "organization" is just GitHub's terminology as I understand it is more an informal collaboration between like-minded ZHA and zigpy developers in order to collect ZHA and zigpy projects ín one place and shared access rights, (noting that the zigpy organization on GitHub has no public members and that you must be a member to see who’s a part of that organization, which I am not myself).

https://github.com/zigpy

Most ZHA related repositories have been moved there over time as a way for like-minded ZHA and zigpy developers that are members of the zigpy organization with yourself included can have shared ownership and access to your zha-toolkit repository and the other projects for collaboration within that community?

Moving it could potentially also help new ZHA developers take notice of this zha-toolkit project.

Hope that you at least will consider this option and if you think it is a good idea then hoping that you all could come to an agreement that will make everyone happy with you continuing this zha-toolkit repository under the zigpy organization together without getting offended by any ownership conflicts.

PS: Same people who are members of that zigpy organization on GitHub also share another GitHub organization called "zha-ng" (short for "ZHA next-generation"?) which contains additional early experimental projects before they became mature to be moved:

https://github.com/zha-ng

**Comment by Hedda** [comment]
@mdeweerd Have you by the way tested zha-toolkit with the new radio API for zigpy so that it does not break backups, etc.? -> https://github.com/zigpy/zigpy/pull/848

_Originally posted by @puddly in https://github.com/zigpy/zigpy/issues/848#issuecomment-1154236530_
> **_Alright, this PR is about as ready as it will ever be. Any final thoughts before I merge this and the corresponding PRs in all of the other radio libraries?_**

**Comment by mdeweerd** [comment]
Short answer: no, and I do not actually have a test setup - I have two systems and both are used for real.

Longer answer: ZHA Toolkit does not rely on the radio interface for backups, but integrates code from the radio specific extensions:
- https://github.com/mdeweerd/zha-toolkit/blob/main/custom_components/zha_toolkit/ezsp_backup.py#L20 (which is essentially a copy of https://github.com/zigpy/bellows/blob/dev/bellows/cli/backup.py#L85 because it could not be called "as is"
- https://github.com/mdeweerd/zha-toolkit/blob/main/custom_components/zha_toolkit/znp.py#L29 which calls https://github.com/zigpy/zigpy-znp/blob/dev/zigpy_znp/tools/network_backup.py#L78 .

Both of these radio specific implementations did not change their CLI implementation, so either they still work, are obsolete and replaced by other methods, or require an update themselves.  In case bellows changes, then zha-toolkit would need an update and I'ld prefer to have a callable function as with ZNP where the code is essentially under `zigpy_znp` control avoiding code repetition.

There may be other things that break with the new API because it is the zha-toolkit use is well beyond the backup feature.

I'll probably discover anything breaking when it's available in HA itself.

**Reply by Hedda** [reply]
> Both of these radio specific implementations did not change their CLI implementation, so either they still work, are obsolete and replaced by other methods, or require an update themselves. In case bellows changes, then zha-toolkit would need an update and I'ld prefer to have a callable function as with ZNP where the code is essentially under `zigpy_znp` control avoiding code repetition.

FYI, there is a PR for zigpy-cli that depends on this change as it in turn will introduce a network backup/restore utility for any adapter implementing the new radio API, see -> https://github.com/zigpy/zigpy-cli/pull/2

There is by the way a matching RFC discission about that new unified radio command line interface for standard backup here -> https://github.com/zigpy/zigpy/issues/557

Oh, and PR https://github.com/zigpy/zigpy-cli/pull/2 will also allow users of ConBee/RaspBee adapters to backup network via zigpy radio library for deconz

**Reply by Hedda** [reply]
@mdeweerd  just an update on this; the new radio API was merged a little more than a month ago and puddly has now also submitted initial patches for ZHA in Home Assistant core (backend) and frontend (GUI) for both automatic and manual Zigbee network backup + restore features/settings;  "***Implements the websocket API for listing, creating, and restoring ZHA network backups. This also adds a ZHA backup platform, which initiates network backup whenever a HA backup is created.***" + "***Add a new tab to the ZHA interface to show the current network settings to allow them to be backed up/restored.***" See pull requests:

https://github.com/home-assistant/core/pull/75791

and

https://github.com/home-assistant/frontend/pull/13295

also see

https://github.com/zigpy/zigpy-cli/pull/13

and

https://github.com/zigpy/zigpy/pull/1006

**Comment by mdeweerd** [comment]
I am sure that a good job was done and I am more than happy to see an implementation in the mainstream offer.  More importantly, restoring should be easier in that setup.

**Comment by Hedda** [comment]
> _**[SUGGESTION] Move zha-toolkit repository to zigpy organization?**_

Bumping request asking to zigpy developers to reconsider merging [mdeweerd](https://github.com/mdeweerd)'s **zha-toolkit** repository/project into the [zigpy](https://github.com/zigpy) organization:

Move https://github.com/mdeweerd/zha-toolkit repo to org -> https://github.com/zigpy/ to get -> https://github.com/zigpy/zha-toolkit

Again, this zha-toolkit provides great value added for advanced ZHA users by allowing use of low-level Zigbee commands through ZHA:

https://github.com/mdeweerd/zha-toolkit/blob/main/README.md

PS: The end-user thread discussing "zha-toolkit" still ranks among the most active in the Zigbee section on the HA community forum:

https://community.home-assistant.io/t/zha-toolkit-toolkit-providing-low-and-high-level-zigbee-commands-through-zha-zigpy/373346

**Comment by Hedda** [comment]
Bumping this again as @mdeweerd who is still alone at maintaining the zha-toolkit repository looks to have paused or stopped doing reviews of pull requests there for about 3-months now, so I assume he might be tired on continuing that as a separate project, and it would probably help if he and his zha-toolkit repository could join the zigpy origination and development team so have other to bounce ideas off, etc.    

Will you please consider/reconsider have zha-toolkit repo moved to the zigpy orginazation?

https://github.com/mdeweerd/zha-toolkit

**Comment by Hedda** [comment]
For reference, raised this again to zha/zigpy devs via this related Open Home Foundation roadmap discussion about ZHA and Zigbee:

* https://github.com/OpenHomeFoundation/roadmap/issues/93

You all might want to post your other ideas/input there anyway as start is a new start at further shaping/improving Zigbee in HA.
